e2e/cmd/e2e.go

```diff
@@ -287,117 +275,106 @@ func waitForKSM(c *k8s.Client, logger *logrus.Logger) (*v1.Pod, error) {
+		se.logger.Infof("deleting release %s", releaseName)
+		err = helm.DeleteRelease(releaseName, cliArgs.Context, se.logger)
 		if err != nil {
-			logger.Errorf("error while deleting release %q", err)
+			se.logger.Errorf("error while deleting release %q", err)
 		}
 	}()
 
 	// At least one of kube-state-metrics pods needs to be ready to enter to the newrelic-infra pod and execute the integration.
 	// If the kube-state-metrics pod is not ready, then metrics from replicaset, namespace and deployment will not be populate and JSON schemas will fail.
-	ksmPod, err := waitForKSM(c, logger)
+	ksmPod, err := se.waitForKSM()
 	if err != nil {
 		return err
 	}
-	return executeTests(c, ksmPod, releaseName, logger, currentScenario)
+	return se.executeTests(ksmPod, releaseName)
 }
 
-func executeTests(
-	c *k8s.Client,
-	ksmPod *v1.Pod,
-	releaseName string,
-	logger *logrus.Logger,
-	currentScenario scenario.Scenario,
-) error {
+func (se *scenarioEnv) executeTests(ksmPod *v1.Pod, releaseName string) error {
 	releaseLabel := fmt.Sprintf("releaseName=%s", releaseName)
 	// We're fetching the list of NR pods here just to fetch it once. If for
 	// some reason this list or the contents of it could change during the
 	// execution of these tests, we could move it to `test*` functions.
-	podsList, err := c.PodsListByLabels(namespace, []string{nrLabel, releaseLabel})
+	podsList, err := se.k8sClient.PodsListByLabels(namespace, []string{nrLabel, releaseLabel})
 	if err != nil {
 		return err
 	}
 
-	logger.Info("Found the following pods for test execution:")
+	se.logger.Info("Found the following pods for test execution:")
 	for _, pod := range podsList.Items {
-		logger.Infof("[%s] status: %s %s", pod.Name, pod.Status.Message, pod.Status.Reason)
+		se.logger.Infof("[%s] status: %s %s", pod.Name, pod.Status.Message, pod.Status.Reason)
 	}
 
-	nodes, err := c.NodesList()
+	nodes, err := se.k8sClient.NodesList()
 	if err != nil {
 		return fmt.Errorf("error getting the list of nodes in the cluster: %s", err)
 	}
-	output, err := executeIntegrationForAllPods(c, ksmPod, podsList, logger)
+	output, err := se.executeIntegrationForAllPods(ksmPod, podsList)
 	if err != nil {
 		return err
 	}
 	var execErr executionErr
-	logger.Info("checking if the integrations are executed with the proper roles")
+	se.logger.Info("checking if the integrations are executed with the proper roles")
 	err = testRoles(len(nodes.Items), output)
 	if err != nil {
 		execErr.errs = append(execErr.errs, err)
 	}
 
-	if currentScenario.ClusterFlavor == minikubeFlavor {
+	if se.scenario.ClusterFlavor == minikubeFlavor {
 		// Minikube use hostPath provisioner for PVCs, which makes kubelet to not report PVC volumes in /stats/summary
 		// See https://github.com/yashbhutwala/kubectl-df-pv/issues/2 for more info.
-		logger.Info("Skipping `testSpecificEntities` because you're running them in Minikube.")
+		se.logger.Info("Skipping `testSpecificEntities` because you're running them in Minikube.")
```

Comment:
Again not quite a fan of the `testSpecificEntities` name, I think we should reflect what the comment above says in the log line (and then possibly remove the comment).

Reply:
Sure, but perhaps out of scope for this already large PR?

Reply:
Yup, you're right. This is already pretty big.